lib/ansible/module_utils/vmware.py

```diff
@@ -902,6 +902,49 @@ def get_managed_objects_properties(self, vim_type, properties=None):
 
         return self.content.propertyCollector.RetrieveContents([filter_spec])
 
+    def generate_user_desired_path_for_vm_folder(self):
+        """
+        Generate the user desired path for vm folder
+        :return: VM folder path
+        """
+        if 'folder' in self.params and 'datacenter' in self.params and \
+                self.params['folder'] and self.params['datacenter']:
+            # Get folder path where virtual machine is located
+            # User provided folder where user thinks virtual machine is present
+            user_folder = self.params['folder']
+            # User defined datacenter
+            user_defined_dc = self.params['datacenter']
+            # User defined datacenter's object
+            datacenter_obj = find_datacenter_by_name(self.content, self.params['datacenter'])
+            # Get Path for Datacenter
+            dcpath = compile_folder_path_for_object(vobj=datacenter_obj)
+
+            # Nested folder does not return trailing /
+            if not dcpath.endswith('/'):
+                dcpath += '/'
+
+            if user_folder in [None, '', '/']:
+                # User provided blank value or
+                # User provided only root value, we fail
+                self.module.fail_json(msg="vmware_guest found multiple virtual machines with same "
+                                          "name [%s], please specify folder path other than blank "
+                                          "or '/'" % self.params['name'])
+            elif user_folder.startswith('/vm/'):
+                # User provided nested folder under VMware default vm folder i.e. folder = /vm/india/finance
+                user_desired_path = "%s%s%s" % (dcpath, user_defined_dc, user_folder)
```

Comment:
We may should check if the user_defined_dc contains a slash and throw an exception a `/` is a valid character for datacenter names, or encode them using urlencode.
I think catching this error is enough for now and the later should be done globally for all modules in the future, as the python and powershell library both suffer from this issue, too.
What do you think?

Reply:
Thank you for your proposal.
I think throw is a good idea.
What about the following exception handling example?

```python
user_defined_dc = self.params['datacenter']
if '/' in user_defined_dc:
    self.module.fail_json(msg="Error Message..........")
```

Reply:
I assume line 916?
That looks good for now ;-)

Reply:
> I assume line 916?

Yes!
Well then, I will add error message handling to the code :)

Reply:
I checked the data center name specifications in the documentation and found that the following characters are also valid in the data center name.
[https://vdc-download.vmware.com/vmwb-repository/dcr-public/6b586ed2-655c-49d9-9029-bc416323cb22/fa0b429a-a695-4c11-b7d2-2cbc284049dc/doc/vim.Folder.html#createDatacenter](https://vdc-download.vmware.com/vmwb-repository/dcr-public/6b586ed2-655c-49d9-9029-bc416323cb22/fa0b429a-a695-4c11-b7d2-2cbc284049dc/doc/vim.Folder.html#createDatacenter)

* %
* \

Therefore, I will `%` and `\` are also add to the conditional expression.